src/Augurk.UI/Services/SearchService.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using Augurk.Entities.Search;
+
+public class SearchService {
+
+    private HttpClient _client;
+
+    public SearchResults LatestResults {
+        get; private set;
+    }
+
+    public SearchService(HttpClient client)
+    {
+        _client = client;
+    }
+
+    public event Func<SearchResults, Task> OnSearchResultsChanged;
+
+    public async Task Search(string searchInput)
+    {
+        if(String.IsNullOrWhiteSpace(searchInput)) {
+            LatestResults = null;
+        }
+        else {
+            var results = await _client.GetFromJsonAsync<SearchResults>($"/api/v2/search?q={searchInput}");
+
+            LatestResults = results;
+        }
+
+        if(OnSearchResultsChanged != null) {
```

Comment:
Is dit de aanbevolen manier om 't te doen met zo'n event? Voelt een beetje lelijk.

Reply:
This is the pattern I could find, based on the single threaded nature of the current Web Assembly.
Perhaps leave it for now, but look for other options with Preview-2 ?

Reply:
Fine